versiebeheer.md

```diff
@@ -0,0 +1,135 @@
+  wijziging,
+* `PATCH` versie betreft het oplossen van *backwards-compatible* issues.
+
+Verder worden een aantal labels gebruikt, achter het versienummer:
+
+* Als er geen aanduiding bij het versienummer staat, gaat het altijd om de
+  meest actuele productie versie van de API.
+* `supported` versies zijn oudere maar ondersteunde versie van de API,
+* `deprecated` versies zijn oudere, niet meer ondersteunde, versies van de API.
+
+Voor meer interne doeleinden kennen we tenslotte nog de volgende labels achter
+het versienummer:
+
+* `RC` (Release Candidate) als de API versie feature-complete is. Alleen
+  kritieke issues worden in deze versie nog verwerkt,
+* `beta` als de versie gereed is voor testers en er geen grote wijzigingen meer
+  verwacht worden. Alleen essentiele zaken worden in dit stadium nog 
+  toegevoegd,
+* `alpha` als de versie nog in ontwikkeling is en er geen enkele garantie is
+  over de werking, stabiliteit e.d.
+
+### Voorbeelden van backwards-incompatible wijzigingen
+
+* Wijzigingen in de URL of fundamentale wijzigingen in de request/response
+  schema's die bij een resource horen,
+* Verwijderen, hernoemen of type-wijziging van een attribuut, query parameter
+  of header,
+* Verwijderen of hernoemen van een resource of API als geheel,
+* Toevoegen van een verplicht attribuut of header.
+
+**LET OP: We spreken altijd over backwards-incompatible wijzigingen vanuit het 
+provider perspectief! Een provider met API op versie `2.1.8` kan altijd goed
+communiceren met een consumer die `2.0.0` t/m `2.1.8` praat (kortgezegd,
+`2.x`) maar het is niet persé waar dat een consumer op versie `2.2.0` goed kan 
+communiceren met een API op versie `2.1.8`.**
+
+### Voorbeelden van backwards-compatible wijzigingen:
+
+* Toevoegen van attributen die optioneel zijn, `null` mogen zijn of een andere
+  standaard waarde hebben.
+* Toevoegen van een element aan een enumeratie,
+* Een niet functionele wijziging in de documentatie,
+* Wijzigingen in de volgorde van attributen.
+
+### Implementaties door leveranciers
+
+We onderscheiden een `consumer` en een `provider`. Een `consumer`, zoals een
+taakapplicatie, maakt gebruik van de API's van een `provider`, zoals de Zaken
+API en/of Documenten API.
+
+Een leverancier implementeert bij voorkeur altijd de meest recente versie van
+een API en zorgt dat haar provider software mee blijft lopen met de actuele 
+versies die VNG aanbiedt, terwijl ook de door VNG ondersteunde versies blijven 
+werken.
+
+Volgens DSO API-24 hebben alle API's de `MAJOR` versie in de URL staan. Om de
+exacte versie te weten wordt de volledige versie als header `API-version`
+meegegeven in de response:
+
+```
+GET /api/v2/zaken/
+
+HTTP 200
+API-version 2.1.8
+...
+```
+
+Als consumers geen versie meegeven in de header wordt de laatste versie 
+gebruikt binnen de `MAJOR` versie die is opgegeven in de URL. Consumers kunnen
+wel een oudere versie afdwingen door deze versie mee te geven als header in het
+request.
```

Comment:
Dat lijkt me nou juist heel onverstandig. Dan wordt het lastig om consumers te dwingen om mee te gaan met de ontwikkelingen en problemen zelf ook op te lossen die in een nieuwere versie van een API opgelost zijn.

Verder denk ik dat als een consumer een oudere versie van een API gebruikt deze een waarschuwing terug moet krijgen en dat dit eigenlijk ook nadat een versie obsolete is niet meer mogelijk moet zijn.

Reply:
Zie hierboven mijn version negotiation voorbeeld.